Adding gdsfactory package

joamatab opened the pull request:
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found some lint.

Here's what I've got...


For **recipes/gdsfactory**:

 * Recipe maintainer "JoaquinMatres" does not exist
 * ``requirements: run: python>=3.7`` must contain a space between the name and the pin, i.e. ``python >=3.7``
 * ``requirements: run: gdspy>=1.6.11`` must contain a space between the name and the pin, i.e. ``gdspy >=1.6.11``
 * If python is a host requirement, it should be a run requirement.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found some lint.

Here's what I've got...


For **recipes/gdsfactory**:

 * Non noarch packages should have python requirement without any version constraints.
 * Non noarch packages should have python requirement without any version constraints.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found it was in an excellent condition.

joamatab commented:
-   - nothing provides requested picwriter 0.5.*
-   - nothing provides requested trimesh 3.9.42.*
-   - nothing provides requested matplotlib-base >=3.5.1,==3.5.*
-   - nothing provides requested rectpack 0.2.2.*
-   - nothing provides requested shapely >=1.8.0,==1.8.*
-   - nothing provides requested phidl 1.6.0.*
-   - nothing provides requested lytest 0.0.20.*

picwriter, new PR needs to be added to conda forge
trimesh, remove pinned version
matplotlib-base, renamed to matplotlib and removed pinned version
rectpack, [new PR](https://github.com/conda-forge/staged-recipes/pull/17632)
shapely, removed new version
phidl is already on conda forge but it is out of date
lytest can be removed as it's only for development

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipes/gdsfactory**:

 * Recipes should usually depend on `matplotlib-base` as opposed to `matplotlib` so that runtime environments do not require large packages like `qt`.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/gdsfactory```) and found it was in an excellent condition.
